lib/libLog.lua

```diff
@@ -0,0 +1,378 @@
+    and success to dt.print()]],
+  Limitations = [[If you use libLog.msg in a callback, the name of the calling routine can't be determined.  A solution
+    is to include some means of reference such as the name of the callback as an argument, i.e. 
+
+      libLog.msg(libLog.debug, "libPlugin.format_combobox:", "value is " .. self.value)
+
+    which would result in
+
+      DEBUG: callback: libPlugin.format_combobox: value is JPEG]],
+}
+
+--[[
+  NAME
+    msg - print a log message
+
+  SYNOPSIS
+    local log = require "lib/libLog"
+
+    log.msg(level, ...)
+      level - table - the type of message, one of: 
+        libLog.debug   - debugging messages
+        libLog.info    - informational messages
+        libLog.warn    - warning messages 
+        libLog.error   - error messages 
+        libLog.success - success messages
+      ... - string(s) - the message to print, which could be a comma separated set of strings
+
+  DESCRIPTION
+    msg checks the level to see if it is enabled, then prints the level type and message if it is.
+    debug messages print straight to the console, info and warn messages to dt.print_error() and error
+    and success to dt.print()
+
+  RETURN VALUE
+    none
+
+  LIMITATIONS
+    If you use libLog.msg in a callback, the name of the calling routine can't be determined.  A solution
+    is to include some means of reference such as the name of the callback as an argument, i.e. 
+
+      libLog.msg(libLog.debug, "libPlugin.format_combobox:", "value is " .. self.value)
+
+    which would result in
+
+      DEBUG: callback: libPlugin.format_combobox: value is JPEG
+
+]]
+
+function libLog.msg(level, ...)
+  local message = {...}
+  if level[2] == true then
+    libLog.print(level[1], libLog.caller(3), unpack(message))
+  end
+end
+
+libLog.libdoc.functions[#libLog.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Return_Value", "Reference"},
+  Name = "print - print the supplied arguments",
+  Synopsis = [[local log = require "lib/libLog"
+
+    log.print(...)
+      ... - arguments to be printed that are converted with tostring()]],
+  Description = [[print loops through the arguments converting each to a string then writing
+    them to stdout.  Spaces are put between each argument on output.]],
+  Return_Value = "none",
+  Reference = "http://stackoverflow.com/questions/7148678/lua-print-on-the-same-line",
+}
+
+--[[
+  NAME
+    libLog.print - print the supplied arguments
+
+  SYNOPSIS
+    local log = require "lib/libLog"
+
+    log.print(...)
+      ... - arguments to be printed that are converted with tostring()
+
+  DESCRIPTION
+    print loops through the arguments converting each to a string then writing
+    them to stdout.  Spaces are put between each argument on output.
+
+  RETURN VALUE
+    none
+
+  REFERENCE
+    http://stackoverflow.com/questions/7148678/lua-print-on-the-same-line
+
+]]
+
+function libLog.print(...)
+  local write = io.write
```

Comment:
I did this because print put tabs between arguments and I needed finer control over the output.  If we use the dictionary and the printing policy, then this should be ok.  Or, I can preprocess the message before I send it to print and not do io.write.  While we're on this, can we change dt.print_error to dt.print_log?  It always says ERROR no matter what your printing out.  This was the reason that I created libLog, so I could separate errors from information.